tool/resources/org/antlr/v4/tool/templates/codegen/Go/Go.stg

```diff
@@ -658,10 +655,10 @@ switch la_ {
 OptionalBlock(choice, alts, error) ::= <<
 p.SetState(<choice.stateNumber>)
 p.GetErrorHandler().Sync(p)
-la_ := p.GetInterpreter().AdaptivePredict(p.GetTokenStream(), <choice.decision>, p.GetParserRuleContext())
+
 <if(alts)>
 
-<alts:{alt | if la_ == <i><if(!choice.ast.greedy)>+1<endif> {
+<alts:{alt | if p.GetInterpreter().AdaptivePredict(p.GetTokenStream(), <choice.decision>, p.GetParserRuleContext()) == <i><if(!choice.ast.greedy)>+1<endif> {
```

Comment:
If you use [this Java.g4 grammar](https://github.com/tunnelvisionlabs/antlr4ts/blob/master/benchmark/Java.g4), you should see an example of the pathological case inside the generated code for the `annotation` rule.